mm2src/coins/z_coin.rs

```diff
@@ -1111,12 +1198,34 @@ impl MarketCoinOps for ZCoin {
     fn my_balance(&self) -> BalanceFut<CoinBalance> {
         let coin = self.clone();
         let fut = async move {
-            let sat = coin
+            let balance_sat = coin
                 .my_balance_sat()
                 .await
                 .mm_err(|e| BalanceError::WalletStorageError(e.to_string()))?;
-            Ok(CoinBalance::new(big_decimal_from_sat_unsigned(sat, coin.decimals())))
+            let spendable = big_decimal_from_sat_unsigned(balance_sat, coin.decimals());
+            let unspendable = {
+                let change_notes = coin
+                    .z_fields
+                    .change_note_db
+                    .sum_changes()
+                    .await
+                    .map_err(|e| BalanceError::WalletStorageError(e.to_string()))?;
+
+                big_decimal_from_sat(change_notes as i64, coin.decimals())
+            };
+
+            if unspendable > spendable {
+                return MmError::err(BalanceError::Internal(
+                    "change must not be greater than balance".to_owned(),
+                ));
+            }
+
+            Ok(CoinBalance {
+                spendable: &spendable - &unspendable,
+                unspendable,
```

Comment:
Yet another remark about the 'unspendable' concept:
Here we include only 'change' into the 'unspendable' balance. But there are also unconfirmed received notes, which we do not track. So users may wonder, why we show the unconfirmed change as 'unspendable' but do not show recently received outputs like that. 
I took a look at the [latest librustzcash version](https://github.com/zcash/librustzcash/blob/7b61fe5025374f7d58d396e449d2b66e3340b749/zcash_client_backend/src/data_api.rs#L143) - they separate unspendable change and received value as different vars in the balance.

Reply:
closing as requires upgrade to latest librustzcash (or modifying it).
Maybe worth adding a new issue like 'show unconfirmed received notes'